app/uk/gov/hmrc/exports/services/reversemapping/declaration/transport/ContainersParser.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.services.reversemapping.declaration.transport
+
+import uk.gov.hmrc.exports.models.declaration.{Container, Seal => SealModel}
+import uk.gov.hmrc.exports.models.StringOption
+import uk.gov.hmrc.exports.services.reversemapping.declaration.{DeclarationXmlParser, XmlParsingException}
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
+import uk.gov.hmrc.exports.services.reversemapping.declaration.XmlTags._
+import uk.gov.hmrc.exports.services.reversemapping.declaration.transport.ContainersParser.{toEitherOfSeq, NO_SEALS}
+
+import scala.xml.NodeSeq
+
+class ContainersParser extends DeclarationXmlParser[Seq[Container]] {
+  override def parse(inputXml: NodeSeq): XmlParserResult[Seq[Container]] = {
+    val transportEquipment = (inputXml \ Declaration \ GoodsShipment \ Consignment \ TransportEquipment).map(parseTransportEquipmentNode)
+    toEitherOfSeq(transportEquipment)
+  }
+
+  private def parseTransportEquipmentNode(inputXml: NodeSeq): Either[XmlParsingException, Container] = {
+    val id = StringOption((inputXml \ ID).text)
+    val eitherSeals = toEitherOfSeq((inputXml \ Seal).map(parseSealNode))
```

Comment:
Why not using `(inputXml \ Seal).map(parseSealNode).toEitherOfList` ?
No need to redefine `toEitherOfSeq`

Reply:
Cool, didn't notice this method tbh :slightly_smiling_face: